service/controller/appcatalog/v1/resource/index/current_test.go

```diff
@@ -115,19 +115,20 @@ func Test_Resource_GetCurrentState(t *testing.T) {
 				t.Fatalf("error == %#v, want matching", err)
 			}
 
-			if tc.expectedConfigMap == nil {
-				if result != nil {
-					t.Fatalf("expected nil ConfigMap got %#v", result)
-				}
-			} else {
-				cm, err := toConfigMap(result)
-				if err != nil {
-					t.Fatalf("error == %#v, want nil", err)
-				}
+			if result != nil && tc.expectedConfigMap == nil {
+				t.Fatalf("expected nil ConfigMap got %#v", result)
+			}
+			if result == nil && tc.expectedConfigMap != nil {
+				t.Fatal("expected non-nil ConfigMap got nil")
+			}
+
+			cm, err := toConfigMap(result)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
 
-				if !reflect.DeepEqual(*cm, *tc.expectedConfigMap) {
-					t.Fatalf("ConfigMap == %#v, want %#v", cm, tc.expectedConfigMap)
-				}
+			if tc.expectedConfigMap != nil && !reflect.DeepEqual(cm, tc.expectedConfigMap) {
```

Comment:
The problem was the comparison for case 1 the expected CM is nil. Also reflect.DeepEqual will compare pointers so you don't need the `*`.